### DISCUSSION TITLE
Potential bug creating animated GIF?  Black/blank section suggests incorrect diff/delta?

### DISCUSSION THREAD

**Discussion by Arakade** [ImageMagick/ImageMagick#1941] [General]
Hi, I'm using IM 7.0.10-6 on Windows, creating an animated GIF using a technique that usually works but is failing for my current images.  Instead of giving a beautiful animation, I'm getting a disappointing (alpha) blank section where it looks like the size of the delta for one of the frames has been calculated incorrectly.  I've attached a zip of my frames (it's for an indie game project and I own the rights -- please don't reuse without asking).  

Here's what I get:

![PickUpPatient](https://user-images.githubusercontent.com/1194266/80320074-203bda80-880c-11ea-98a2-7d0a59a40e2d.gif)

Here's the command I'm using (duplicated in [tmp.bat.txt](https://github.com/ImageMagick/ImageMagick/files/4536493/tmp.bat.txt) for convenience):

magick Background.png -dispose previous -delay 20 5.png 6.png 7.png 8.png 9.png 8.png 7.png 6.png -coalesce -delete 0 -layers CompareAny -loop 0 PickUpPatient.gif

And here are the input images: [PickUp.zip](https://github.com/ImageMagick/ImageMagick/files/4536490/PickUp.zip)

Any thoughts?  I'll admit I'm no expert with the mighty ImageMagick -- I was quite pleased I got it to do what I'd hoped previously but now I'm worried I'm doing something wrongly.  I've used this approach fine with other sequences of images.  I've tried tweaking these images but get the same result?!  Any advice appreciated.  Thanks!

Version info:

Version: ImageMagick 7.0.10-6 Q16 x64 2020-04-06 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

**Comment by fmw42** [comment]
Where is background.png? It is not in the zip file.

**Comment by Arakade** [comment]
Whoops, my apologies @fmw42 !  Attached : ![Background](https://user-images.githubusercontent.com/1194266/80321347-541afe00-8814-11ea-8d8b-3b35b09823d4.png)

I've continued reading and fiddling and have found that using -layers optimize instead of -layers CompareAny produces a correct albeit larger animation.  Obviously if there's a problem with CompareAny, should I report it somewhere (or can I get a link to a bug so I can follow, please)?

Thanks!

**Comment by fmw42** [comment]
Try this:

Unix syntax:

```
magick -dispose previous -delay 20 \
background.png null: 5.png 6.png 7.png 8.png 9.png 8.png 7.png 6.png \
-compose over -layers composite \
-layers optimize \
-loop 0 \
PickUpPatient.gif

```

Windows Syntax:

```
magick -dispose previous -delay 20 ^
background.png null: 5.png 6.png 7.png 8.png 9.png 8.png 7.png 6.png ^
-compose over -layers composite ^
-layers optimize ^
-loop 0 ^
PickUpPatient.gif

```

![PickUpPatient](https://user-images.githubusercontent.com/13458218/80321680-cd463100-87d3-11ea-9dba-4030d1c55c92.gif)
